add newer_than parameter to SFTP sensor

AdamPaslawski opened the pull request:
closes: #21655
Add a parameter newer_than to the sftp sensor to allow for specifying a datetime for which the file in the SFTP site should be newer than

AdamPaslawski committed:
add newer_than parameter to SFTP sensor (#21655)

AdamPaslawski commented on airflow/providers/sftp/sensors/sftp.py:
```diff
@@ -54,4 +63,9 @@ def poke(self, context: 'Context') -> bool:
+        if self.newer_than:
+            _mod_time = datetime.strptime(mod_time, '%Y%m%d%H%M%S')
+            _newer_than = make_naive(self.newer_than) if not is_naive(self.newer_than) else self.newer_than
+            return _newer_than <= _mod_time
```
I decided to assume a naive comparison and allow the sensor to be configured with an offset aware or unaware newer_than parameter, thoughts on this @eladkal ?

potiuk replied:
I think you shoudl convert all dates to UTC. It's surprisingly hard to find what is the timezone of st_mtime but all signs point at UTC.

AdamPaslawski replied:
I agree. Assuming the SFTP site is always on the same time-zone as the airflow instance is weaker than assuming UTC.

I'll make that change.

jbampton committed:
Minor `pre-commit` config clean up (#21794)

dstandish committed:
Airflow `db downgrade` cli command (#21596)

josh-fell committed:
Add dbt Cloud provider (#20998)

kazanzhy committed:
Suppress hook warnings from the Bigquery transfers (#20119)

h2cone committed:
Add 'method' to attributes in HttpSensor. (#21831)

Jolcik committed:
Fix DAG date range bug (#20507)

pbabics committed:
Fix handling of empty (None) tags in `bulk_write_to_db` (#21757)

rsg17 committed:
Quick Update GCS Presto (#21855)

mariotaddeucci committed:
Add JSON output on SqlToS3Operator (#21779)

Commit:
Add 'Show record' option for variables (#21342)

alexchen8 committed:
Configurable AWS Session Factory (#21778)

hterik committed:
Fix mixup up shadowed ImportError in db_cleanup (#21862)

TableConfig requires models.ImportError, while the try/except
on line 128 requires the builtin ImportError

This was causing errors when running with k8 scheduler,
without having celery packages installed.

humit0 committed:
update value from database to celery workers (#21859)

Co-authored-by: huan.15 <huan.15@kakaocorp.com>

potiuk committed:
Allow to switch easily between Bullseye and Buster debian versions (#21546)

We are preparing to switch from Buster to Bullseye and this is
the second change that is needed (following #21522). This change
allows to choose whether we want to use Buster or Bullseye images
as a base. We need to be able to choose, because:

1) we want to keep backwards compatibility and continue our
   users to build Buster-base images
2) we cannot yet fully switch to Bullseye because MsSQL's odbc
   driver does not yet support Bullseye and we reached out to
   mysql maintainers to learn about their plans to make the
   decision on when and how we are going to support Bullseye and
   MSSQL.

   Details of this discussion are in:
   https://github.com/MicrosoftDocs/sql-docs/issues/7255#issuecomment-1037097131

This PR adds the capability of choosing the DEBIAN_VERSION in
Breeze when building images but does not yet switch from Buster to
Bullseye

potiuk committed:
Switch to Debian 11 (bullseye) as base for our dockerfiles (#21378)

Debian 11 Bullseye have been released some time ago as the new
LTS Debian release and already all our dependencies (including
MySQL and MSSQL ODBC drivers) caught up with it so we can finally
migrate to it.

This change switches base images to bullsey for our Dockerfiles
as well as for Redis image we are using in CI.

The relevant packages have been updated to include that
and documentation have been updated.

Examples of ours also are updated to use "bullseye" rather than
buster.

Closes: #18190
Closes: #18279

ashb committed:
Revert "Switch to Debian 11 (bullseye) as base for our dockerfiles (#21378)" (#21874)

This reverts commit 5d89dea56843d7b76d5e308e373ba16ecbcffa77.

The issue is not a random IO timeout -- it's a problem with the file in the repo.

Reverting this right now as all PRs are failing :(

potiuk committed:
Update to latest version in Dockerfile (and add instructions) (#21876)

malthe committed:
Log traceback in trigger excs (#21213)

bbovenzi committed:
Make Grid and and Graph view work with task mapping (#21740)

* Expand mapped tasks in the Scheduler

Technically this is done inside
DagRun.task_instance_scheduling_decisions, but the only place that is
currently called is the Scheduler

The way we are getting `upstream_ti` to pass to expand_mapped_task is
all sorts of wrong and will need fixing, I think the interface for that
method is wrong and the mapped task should be responsible for finding
the right upstream TI itself.

* make UI and tree work with mapped tasks

* add graph tooltip and map count

* simplify node label redraw logic

* add utils.js and map_index to /taskInstances

* use TaskInstanceState instead of strings

* move map_index on /taskinstance to separate PR

* check to use Task or Tasks

* remove `no_status` and use TaskInstanceState

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>

jedcunningham committed:
Fix UPDATING section on SqlAlchemy 1.4 scheme changes (#21887)

santonaastronomer committed:
Update databricks.rst (#21886)

The method outlined in the current doc results in a 403 error, which can be avoided by following Databricks' documentation on this topic: https://kb.databricks.com/dev-tools/invalid-access-token-airflow.html. The changes suggested here reflect the Databricks doc.

kaxil committed:
Fix the Type Hints in ``RedshiftSQLOperator`` (#21885)

`sql` accepts `Union[str, Iterable[str]],` not `Optional[Union[Dict, Iterable]],`

Commit:
Allow templates in more DataprocUpdateClusterOperator fields (#21865)

dstandish committed:
Autogenerate migration reference doc (#21601)

* document airflow version in each alembic migration module and use this to autogen the doc
* update each migration module to have the same description used in migration ref (so it can be used in autogen)

mixilchenko committed:
Fix filesystem sensor for directories (#21729)

Fix walking through wildcarded directory in `FileSensor.poke` method

pingzh committed:
add celery.task_timeout_error metric (#21602)

* log celery task id to correlate logs

* add celery.task_timeout_error metric

schirag1993 committed:
Feature: Add invoke lambda function operator (#21686)

haoli-buaa committed:
Update modules_management.rst (#21889)

need to add packages in setup.py, otherwise the package can not be found

potiuk committed:
Switch to Debian 11 (bullseye) as base for our dockerfiles (#21378) (#21875)

Debian 11 Bullseye have been released some time ago as the new
LTS Debian release and already all our dependencies (including
MySQL and MSSQL ODBC drivers) caught up with it so we can finally
migrate to it.

This change switches base images to bullsey for our Dockerfiles
as well as for Redis image we are using in CI.

The relevant packages have been updated to include that
and documentation have been updated.

Examples of ours also are updated to use "bullseye" rather than
buster.

Closes: #18190
Closes: #18279

uranusjr committed:
Fix TaskDecorator type hints (#21881)

ashb committed:
Change BaseOperatorLink interface to take a ti_key, not a datetime (#21798)

potiuk committed:
Limits Yandexcloud to 0.145 to unblock main failing tests. (#21899)

The yandexcloud 0.145 broke logging feature of the client library
and it fails yandex provider tests.

https://github.com/yandex-cloud/python-sdk/issues/47

potiuk committed:
Fix Kubernetes example with wrong operator casing (#21898)

The Kubernetes example of ours had bad casing in "In" operator
("in") and recent kubernetes client library started to fail importing
the example).

This PR fixes case for the operator.

avkirilishin committed:
EdgeModifier refactoring (#21404)

dstandish committed:
Deprecate non-JSON conn.extra (#21816)

Connection extra field is generally assumed to be JSON but we don't actually require it.  Here we deprecate non-JSON extra so that in 3.0 we can require it.  Further, we require that it not just be any json but must also parse as dict, because a string value such as '"hi"' or '[1,2,3]' is json, but a very bad practice.

potiuk committed:
Limit yandexcloud to < 0.142 (#21903)

It turned out the yandexcloud bugfix attempt from #21899 was not
successful. The bug was present already in 0.142

potiuk committed:
Fix handling some None parameters in kubernetes 23 libs. (#21905)

Kubernetes 23.* is more picky when it comes to values passed to
Pod Generator - it requires:

* imagePullPolicy
* dnsPolicy
* restartPolicy

to be not None.

We are fixing it in the way, that we simply skip setting those
if they are None.

Commit:
Make project_id argument optional in all dataproc operators (#21866)

`DataprocHook` handles `project_id=None` in `@GoogleBaseHook.fallback_to_default_project_id` decorator. In some operators `project_id` was already optional. Make it optional for all.

jedcunningham committed:
Update list of non-core files/paths (#21907)

akifcakir committed:
Add-showing-runtime-error-feature-to-DatabricksSubmitRunOperator (#21709)

ashb committed:
Refactor KubernetesPodOperator tests (#21911)

This re-works many of the KubePodOp tests to just _real_
DAGs/DagRun/TaskInstance rather than faked ones, and splits out some of
the tests that checked 3 or 4 "concepts" in to multiple separate test
functions

potiuk committed:
Add Python 3.9 support to Hive (#21893)

Hive support for Python 3.9 has been removed in #15515 but cloudera
released new ssl 0.3.1 library version to support it and we should
be able to get Hive provider working for Python 3.9 too.

Fixes: #21891

jedcunningham committed:
Chart: move updating note about config removal (#21919)

This was accidentally placed in the 1.4.0 section, which is already
released. Move it to the 1.5.0 section where it should be.

kaxil committed:
Unpin ``pandas-gbq`` and remove unused code (#21915)

* Unpin ``pandas-gbq`` and remove unused code

`BigQueryPandasConnector` was previously used by `BigqueryHook.get_pandas_df`. This was fixed in https://github.com/apache/airflow/commit/ad308ea441372f2b44b4292c3779eb745f2ed48c (**in 2018**). However we forgot to remove `BigQueryPandasConnector` which is age-old code (2016) and used private methods.

2016 code - https://github.com/apache/airflow/pull/1452/files

kaxil committed:
Unpin `google-cloud-memcache` (#21912)

josh-fell committed:
Add `test_connection` method to `AzureDataFactoryHook` (#21924)

sungpeo committed:
py files doesn't have to be checked is_zipfiles in refresh_dag (#21926)

millin committed:
Fix incorrect data provided to tries & landing times charts (#21928)

uranusjr committed:
Ensure deps is set, convert BaseSensorOperator to classvar (#21815)

uranusjr committed:
Only allow mapping against return value XCom (#21930)

jedcunningham committed:
Fix single backticks in rst changelogs (#21922)

ashb committed:
Add map_index to pods launched by KubernetesExecutor (#21871)

I also did a slight drive-by-refactor (sorry!) to rename `queued_tasks
and `task` inside `clear_not_launched_queued_tasks` to `queued_tis` and
`ti` to reflect what they are.

ashb committed:
Add map_index label to mapped KubernetesPodOperator (#21916)

Commit:
update freshworks company users (#21932)

ferruzzi committed:
Update ECS sample DAG and Docs to new standards (#21828)

ephraimbuddy committed:
Chart: 1.5.0 changelog (#21906)

Add changelog for chart version 1.5.0.

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

dstandish committed:
Rename `xcom.dagrun_id` to `xcom.dag_run_id` (#21806)

We use snake case for tables and columns in the metadata database.  So references (in the database) to `dag_run.id` should be `dag_run_id`.  Historically though, in many places in the codebase we used the expression "dag run id" to refer to the `run_id` attribtue of the DagRun model.  So here we try to disambiguate this where necessary by using `run_id` to refer to DagRun.run_id and `dag_run_id` to refer to `DagRun.id`.

One area where I did not make any changes is in the REST API which exposes parameter `dag_run_id` to refer to the `run_id` field in DagRun.  Since this param is user-facing, it's OK for it to be disconnected from the internal code references and certainly from database object naming.  We can consider renaming to `run_id` in the next major API release.  Also did not change the code in `views.py`.

Squashed commits:

* rename IN_MEMORY_DAGRUN_ID
* rename ARG_EXECUTION_DATE_OR_RUN_ID
* rename DagrunIdDep
* fake_dagrun_id
* rename dagrun_id to dag_run_id
* clarify docstrings / comments
* rename dag_run_id to run_id in dag model
* _set_dag_run_state param dag_run_id -> run_id

potiuk committed:
Exit with appropriate message for shell in PROD image in breeze (#21848)

While Breeze allows to build production image as well as the CI
image, it is not really meant to be used to run production image.

Unlike CI image, which requires many parameters to pass to docker
and docker-compose commands, most of the docker commands for PROD image
are already possible and easy to be executed with regular docker
command.

When someone tries to execute the PROD image they will get appropriate
error message and link to the documentation explaining how to
run commands in production image.

Fixes: #21723

josh-fell committed:
Make container creation configurable when uploading files via WasbHook (#20510)

john-jac committed:
Add RedshiftDataHook (#19137)

Use the AWS `redshift-data` API to interact with AWS redshift clusters

Co-authored-by: john-jac <jacnjoh@amazon.com>

dstandish committed:
Add docs for `db upgrade` / `db downgrade` (#21879)

potiuk committed:
Add CI jobs and tooling to aid with tracking backtracking pip issues (#21825)

* Add CI jobs and tooling to aid with tracking backtracking pip issues

This is a follow-up after investigation done with failing main
builds (resulting in #21824 and tracked in
https://github.com/pypa/pip/issues/10924)

Handling failure of image building has been also improved as part
of the PR. Instead of separate "cancel" job (which did not really
work anyway) we build and push empty images instead and the
empty images are handled in the "wait for images" job
with appropriate message.

* Update dev/breeze/src/airflow_ci/find_newer_dependencies.py

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

uranusjr committed:
Enhance magic methods on XComArg for UX (#21882)

* Enhance magic methods on XComArg for UX

Implementing Python's magic methods for purposes not matching their
original design can have unwanted consequences due to default
interactions between those methods. This implements a custom __iter__ so
the custom __getitem__ does not trigger the default __iter__ behavior
and result in bad run-time behavior.

Bowrna committed:
enter the shell breeze2 environment (#21145)

potiuk committed:
Proposed policy for Provider's minimum supported version. (#21696)

This is the proposed policy, where we explain when by default
we increase the minimum supported version for the providers we
release from the community.

bhavaniravi committed:
Add detailed email docs for Sendgrid (#21958)

Improved documentation for Sendgrid email

Fixes #20419

ephraimbuddy committed:
Dev: Update script used in generating issues for status of testing of RC (#21950)

caxefaizan committed:
Fix typo in docs (#21964)

ephraimbuddy committed:
Remove unnecessary loggings from offline sql generation command (#21962)

I found a way to remove unnecessary loggings that comes from validating revisions in the offline
migration command. Also cleaned up some codes

ashb committed:
Change KubePodOperator labels from exeuction_date to run_id (#21960)

Now that execution_date isn't the PK for TaskInstance we should replace
it with run_id.

There is on backwards compatibility concern here, as these labels are
only needed for re-attaching to a container, and the way we get the
run_id is compatible back to 2.1

uranusjr committed:
More explicit mapped argument validation (#21933)

* More explicit mapped argument validation

Instead of always using MagicMock to validate mapped arguments, this
implements a more sophisticated protocol that allows an operator to
implement a 'validate_mapped_arguments' to provide custom validation
logic. If an operator just wants to use __init__ for validation,
however, they can set a flag 'mapped_arguments_validated_by_init' to get
the behavior easily. (This does *not* use MagicMock, however, since any
custom validation logic should be able to handle those on its own).

The 'validate_mapped_arguments' flag is currently only set on
PythonOperator. It can likely be used on a lot more operators down the
road.

* Add flag to distinguish a validation-only init

There's just too much magic during a task's initialization that tries to
add it into the dependency graph. This flag is needed to work around all
that, I think.

ashb committed:
Ensure that `airflow dags tests` works for mapped DAGs (#21969)

uranusjr committed:
Add compat shim for SQLAlchemy to avoid warnings (#21959)

bbovenzi committed:
replace extra links value (#21971)

jedcunningham committed:
Run inclusive language check on CHANGELOG (#21980)

We should run our inclusive language check on our changelog. We only
have a single failure when we do, ironically from when we put the
check in place.

jedcunningham committed:
Fix changelog/updating typos (#21979)

Some of these are just British vs American spelling differences and
names, others are true typos.

treyyi committed:
Rename 'S3' hook name to 'Amazon S3' (#21988)

PR to produce more intuitive and organized names for Amazon hooks.

potiuk committed:
Change the storage of frame to use threadLocal rather than Dict (#21993)

There is a very probable WeakKeyDict bug in Python standard
library (to be confirmed and investigated further) that
manifests itself in a very rare failure of the
test_stacktrace_on_failure_starts_with_task_execute_method

This turned out to be related to an unexpected behaviour
(and most likely a bug - to be confirmed) of WeakKeyDict
when you have potentially two different objects with the
same `equals` and `hash` values added to the same
WeakKeyDict as keys.

More info on similar report (but raised for a bit different
reason) bug in Python can be found here:

https://bugs.python.org/issue44140

We submitted a PR to fix the problem found 
https://github.com/python/cpython/pull/31685

ephraimbuddy committed:
Add test to run DB downgrade in the CI (#21273)

This attempts to add db upgrade/downgrade test to the CI

dstandish committed:
Simplify tests for CLI connections commands (#21983)

The tests (particularly with the export subcommand) were doing a fair bit of unnecessary mocking. In particular they mocked builtins.open. But there's no good reason that we can't actually write out the file and read the file in the assert (which is a much better way to test this). There were also a number of assertions re splittext which are not really relevant.

I suspect the reason that the author did this is because the file wasn't closed, so when you read the value from the exported file (if the process has not been exited) then the buffer may not be fully written to disk. I fix this by closing the file after writing the connections.

As part of this change I also update to use current pytest techniques for parameterization and temp files.

blag committed:
Add per-DAG delete permissions (#21938)

This PR adds per-DAG delete permissions and extends the sync-perms subcommand to add delete permissions to the database for all existing DAGs (it does not, however, grant any of the new DAG delete permissions to any roles).

dstandish committed:
Don't validate that Params are JSON when NOTSET (#22000)

If params are NOTSET, then JSON validation will of course fail.

jedcunningham committed:
Minor cleanup on Celery config (#21880)

This rewords the description and changes the type of
``worker_enable_remote_control`` and sets a default for
``worker_prefetch_multiplier`` instead of relying on fallback.

dstandish committed:
Add brief examples of integration test dags you might want (#22009)

kazanzhy committed:
Upgrade `moto` library to version 3.0 (#22005)

Update `moto` library to version 3.0

subkanthi committed:
Local kubernetes executor (#19729)

dstandish committed:
Resolve mypy issue in athena example dag (#22020)

* Resolve mypy issue in athena example dag

* fixup! Resolve mypy issue in athena example dag

frankcash committed:
Updates FTPHook provider to have test_connection (#21997)

* Updates FTP provider to have test_connection

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

shahkshitij15 committed:
Add autodetect arg to external table creation in GCSToBigQueryOperator (#21944)

* Added autodetect parameter in external table creation in GCSToBigQueryOperator

xissy committed:
Fix typo in docs (#22024)

potiuk committed:
Remove back the limitation of Yandex as a bug in 0.142 is fixed (#22023)

The original bug in Yandex causing test failures has been
fixed, so we are removing the upper-bound limit for it.

potiuk committed:
Fix failing main after merging SFTP hook test connection (#22026)

The #21997 implemented test_connection method in FTP hook, but
we were using FTP hook to actually test .... what happens
when the test_connection method was missing :)

Unfortunately the test for it was not run because the "core"
tests are skipped in case only provider change :(.

This PR switches the test_connection missing test to use GRPC hook
(it does not have the test_connection method) and moves the whole
test_connection.py to "always" category of tests that are
always executed - there are more tests there that have some
assumptions on the existing connections and having the tests
executed "always" makes sense to avoid similar problems in the
future.

kazanzhy committed:
Added AWS RDS operators (#20907)

kazanzhy committed:
Added AWS RDS sensors (#21231)

khalidmammadov committed:
Fixing bug when roles list is empty (#18590)

Bowrna committed:
added docker network_mode options (#21986)

potiuk committed:
Removes limitations from Dask dependencies (#22017)

Dask dependencies were holding us back - when it comes to upgrading
somoe of the packages (for example apache-beam and looker - in google
provider). This PR removes the limitations but with a twist.

* Dask tests stop working. We reach out to the Dask Team to fix them
  but since a very old version of `distributed` library was used
  the Dask team is called for help to fix those

* The typing-extensions library was limited by `distributed` but it
  seems that version 4.0.0+ breaks kubernetes tests

vincbeck committed:
Move S3ToRedshiftOperator documentation to transfer dir (#21975)

rustikk committed:
Default args type check (#21809)

vincbeck committed:
Cleanup RedshiftSQLOperator documentation (#21976)

EricGao888 committed:
Switch oss hook tests in alibaba-provider to use Mocks (17617) (#21992)

subkanthi committed:
Change default python executable to python3 for docker decorator (#21973)

Added support to pass python executable(python3) to Docker decorator.

zachliu committed:
retry on very specific eni provision failures (#22002)

o-nikolas committed:
Add docs and sample dags for AWS Batch (#22010)

kaxil committed:
Dev: Allow easy repacking of Providers (#22018)

* Dev: Allow easy repacking of Providers

This PR adds `--skip-tag-check` to easily repackage a provider by running:

```
rm -rf provider_packages/airflow
cp -r airflow provider_packages
mkdir dist
python dev/provider_packages/prepare_provider_packages.py generate-setup-files --skip-tag-check "cncf.kubernetes"
python dev/provider_packages/prepare_provider_packages.py build-provider-packages --skip-tag-check "cncf.kubernetes"
```

This allow us to repackage providers without using Docker containers/breeze or git.

potiuk committed:
Fix mypy errors resulting from typing updates in marshmallow (#22044)

Marshmallow library has now better typing and mypy detected
a potential problem with messages not always being dict in
ValidationError. Switched to normalized_messages to fix it.

potiuk committed:
Update limits of dependencies after `dask` test disabling (#22046)

Some of the tests failed previously with typing extensions above 4.
This PR attempts to relax the limit and check if the problems
still appear.

Also new tests (S3) started to fail when a new `responses` library
version has been released today.

So this change also add limits to the responses library in order
to make sure the tests pass.

Issue https://github.com/getsentry/responses/issues/511 has been
opened to raise it to `responses` library maintainers.

caxefaizan committed:
missing quotes (#21990)

Missing quotes throws the following error:
```
airflow:
- env.0.value: Invalid type. Expected: string, given: boolean
```

potiuk committed:
Add information on DAG pausing/deactivation/deletion (#22025)

Many of our users do not understand how DAG deactivation, deletion
works - it's quite straightforward for us, who understand how
scheduler, serialization and refresh works, but we have not really
documented everywhere what the differences are between pausing,
deactivation and actual deletion of the metadata.

It has been distributed in a few places (API documentation, comments
when you issued a "delete" UI action, but I found that there is no
single place where it is described.

This PR adds documentation in "DAG" page about it - explaining
what happens in those different states and instructing the user
to follow -> remove dag file -> wait for deactivation -> delete
pattern if they want to remove a DAG.

Related: #21864

alekseiloginov committed:
Add Looker PDT operators (#20882)

NivekNey committed:
Update best-practices.rst (#22053)

jbampton committed:
Fix spelling (#22054)

potiuk committed:
Add documentation for Feb Providers release (#22056)

This provider's release is pretty special as we release
all providers with Python 3.10 support.

jedcunningham committed:
DB upgrade is required when updating Airflow (#22061)

Just strengthen the language that it is "required", not "recommended" to
run `airflow db upgrade` when upgrading Airflow versions.

vincbeck committed:
Add sample dag and doc for RedshiftToS3Operator (#22060)

patricker committed:
Allow for uploading metadata with GCS Hook Upload (#22058)

jbampton committed:
Clean up the `pre-commit` config (#22052)

ferruzzi committed:
Dynamo to S3 Sample DAG and Docs (#21920)

easontm committed:
Add example config of sql_alchemy_connect_args (#22045)

I saw a few people in slack confused about how this setting is applied. After verifying this behavior for my own deployment, I'm including it in the docs.

victorphoenix3 committed:
fixes query status polling logic (#21423)

eladkal committed:
Add template fields to DynamoDBToS3Operator (#22080)

alexott committed:
Add new options to DatabricksCopyIntoOperator (#22076)

This includes:
* `encryption` - to specify encryption options for a given location
* `credential` - to specify authentication options for a given location
* `validate` - to control validation of schema & data

mhenc committed:
Store callbacks in database if standalone_dag_processor config is True. (#21731)

norm committed:
Add Mapped task instance endpoint (#21965)

Add an API endpoint that supplies details of a mapped task instance.

victorphoenix3 committed:
refactors polling logic for athena queries (#21488)

rsg17 committed:
Bug-fix GCSToS3Operator (#22071)

Commit:
Use yaml safe load (#22085)

potiuk committed:
Improve guidelines for contributors for provider testing (#22087)

The instructions on how to test providers are improved
and added to an issue template.

potiuk committed:
Revert "Use yaml safe load (#22085)" (#22089)

This reverts commit 7f4935bab36c41d5927610e38c46a30da2b80906.

subkanthi committed:
Fixed dask executor and tests (#22027)

 Fixed dask executor and tests, distributed package does not ship with tests folder and the certificates, added certificates to certs folder

Wilderone committed:
Fix/21994 liveness probe (#22041)

ephraimbuddy committed:
Update changelog for helm chart 1.5.0 (#22090)

ephraimbuddy committed:
Update release date on changelog (#22094)

Commit:
Use yaml safe load (#22091)

bbovenzi committed:
Pause auto-refresh when page is hidden (#21904)

* pause autorefresh when page is hidden

* fix tests

* use react query and simplify webAPI vars

* use initRefresh instead of always starting refresh

ephraimbuddy committed:
Add `list-import-errors` to `airflow dags` command (#22084)

This will help users to see the dags with import error and enable scripts
process the output

NiloFreitas committed:
Add guide for DataprocInstantiateInlineWorkflowTemplateOperator (#22062)

ferruzzi committed:
EMR on EKS Sample DAG and Docs Update (#22095)

potiuk committed:
Upgrade PIP to 22.0.4 (just released) (#22081)

steveyz-astro committed:
Add new map_index field to log message (#22098)

ephraimbuddy committed:
Show import error for 'airflow dags list' CLI command (#21991)

When there's an import error in a dag, the dag doesn't show up in the list of dags
and there's no indication that there's an import error.
This PR fixes that

eladkal committed:
bump sphinx-jinja (#22101)

steveyz-astro committed:
If uploading task logs to S3 fails, retry once (#21981)

alexchen8 committed:
Fix typo in AWS doc (#22097)

jbampton committed:
Fix spelling (#22107)

eladkal committed:
bump moto version (#22099)

edithturn committed:
Replace build image bash script  in CI with build image in python (#22008)

Use Breeze2 build image python on CI

uranusjr committed:
Rename task-mapping trigger to 'expand' (#22106)

potiuk committed:
Add Codespaces support (#22082)

* Add Codespaces support

Codespaces is (soon to be GA) feature that allows to start
development environment on a virtual machine directly from the
GitHub UI or from VSCode connected with GitHub account.

This PR adds configuration that allows to start codespaces
environment (based on Breeze's docker compose environment)
in the basic form that allows the users to run basic unit
tests.

It does not replace Breeze/CI but for most users who want
to just run a few tests, make sure that all dependencies
and plugins are installed so that the users can get the
VSCode autocomplete and execution of the tests possible.

This is a great way to begin your journey with Airflow - while
the Codespace machines are not very powerful and the integration
tests might require more resources, the "no-setup"
environment and no need to have powerful machine is great
for contributors who cannot afford powerful development
workstations.

* Update CONTRIBUTING.rst

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* Update scripts/ci/libraries/_build_images.sh

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* Update CONTRIBUTING.rst

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

ashb committed:
Add map_index to RenderedTaskInstanceFields (#22004)

In order to do this "properly" we have also migrated it from using
execution_date to run_id in its PK

ashb committed:
Require SQLAlchemy 1.4 (#22114)

Some of the exceptions have moved between 1.3 and 1.4
(`sqlalchemy.orm.exc import MultipleResultsFound` becomes `from
sqlalchemy.exc import MultipleResultsFound`)

Since we only ever test with the "latest" version we should update the
requirement to be tighter to ease upgrades.

(Without this the UI/API will fail to start)

sbakiu committed:
Update logging-tasks.rst (#22116)

Fix PYTHONPATH Environment Variable name

potiuk committed:
Use "dev" from main in "build-images" workflow (#22119)

ashb committed:
Allow searching/filtering Browse Task Instances view by map_index (#22117)

ashb committed:
Don't try to create automigration for celery tables (#22120)

uranusjr committed:
Fix RTIF test against the apply-expand (#22121)

mirobertod committed:
Add webserver PodDisruptionBudget (#21735)

potiuk committed:
Add support for ARM platform (#22127)

This support is mostly for the developers, not for CI full chain yet.
It has several limitations:

* no MySQL client support
* no MsSQL client support
* no CI tests yet

What is implemented:

* automated detection of ARM/AMD architecture when building and
  running breeze
* automated cache refresh on CI for ARM/AMD

Currently only development (ghcr.io) images are supported for ARM.

Fixes: #18849
Fixes: #17494
Relates to: #15635

The images published in DockerHub for now are AMD64 only. We will
run development with M1 images for some time and later we will
likely make our DockerHub images multi-platform as well.

Also Hadolint does not have ARM images yet so we had to disable it
and we should re-enable it back after the support is added.
See https://github.com/hadolint/hadolint/issues/411

potiuk committed:
Rename wrongly named script for pushing prod images (#22139)

potiuk committed:
Switches backend to sqlite when building images (#22140)

The backend is now checked at execution so we should not
set it to postgres when postgres DB is not started.

potiuk committed:
Fix location of --push directive for prod images (#22142)

Previous #22127 had `--push` added in a wrong place.
It should only be added when cache is being built and it
was added always, with resulted in authentication error as
login has not been performed before the --push.

josh-fell committed:
Add more template fields to `DbtCloudJobRunOperator` (#22126)

josh-fell committed:
Pass explicit overrides in `DbtCloudJobRunOperator` to `DbtCloudHook` (#22136)

budney committed:
Explicitly adding read permission to Docker entrypoint. (#22135)


Co-authored-by: Len Budney <Len.Budney@grantstreet.com>

andrzejzysko committed:
Update cloud_composer.rst (#22124)

Jorricks committed:
Add queue button to click-on-DagRun interface. (#21555)

* Initial implementation of adding Queue button to DagRun interface

* Implement the test cases

* FIX Add all required MyPy ignores

* FIX import

* Update airflow/www/views.py

FIX Documentation

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

* update modal UI

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

bbovenzi committed:
Reduce rerendering of TaskName component (#22156)

bbovenzi committed:
Reduce rerendering of component (#22161)

Memoize component with a custom compare function to reduce rerenders

ashb committed:
Don't try to expand mapped tasks if they can't run (#22155)

This lead to the case where we tried to expand a mapped task when its
upstream was failed, but we hadn't yet set the (unexpanded) mapped TI to
UPSTREAM_FAILED.

In order to support this behaviour we need to add a new "ignore" flag to
DepContext -- not the best pattern, but we should follow it for now

dstandish committed:
Order filenames for migrations (#22168)

Sometimes you want to quickly find recent migrations.  This makes it easier

zachliu committed:
Bugfix for retrying on provision failuers(#22137)

potiuk committed:
Conditional skipping of provider tag when it exists (#22172)

dstandish committed:
Fix ordering of migration filenames (#22169)

ephraimbuddy committed:
Use jobs check command for liveness probe check in airflow 2 (#22143)

This PR removes the current use of python code for liveness probe check
commands in the Scheduler & Triggerer deployments

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

094459 committed:
additional information in the ECSOperator around support of launch_type=EXTERNAL (#22093)

* additional information in the ECSOperator around support of launch_type=EXTERNAL

vincbeck committed:
Fix RedshiftDataOperator and update doc (#22157)

pingzh committed:
Add pip_install_options to PythonVirtualenvOperator (#22158)

EricGao888 committed:
Switch unit tests for oss operator and oss sensor in alibaba provider to use mocks (#17617) (#22178)

EricGao888 committed:
Add oss_task_handler into alibaba-provider and enable remote logging to OSS (#21785)

bbovenzi committed:
Improve StatusBox rendering in Grid view (#22170)

* reduce grid task instance rerenders

* fix test

uranusjr committed:
Add map_index to XCom model and interface (#22112)

* Add map_index to XCom primary key

This is not actually stored correctly yet. We still need to fix the XCom
interface.

* Add map_index to XCom interface

This adds an additional (optional) map_index argument to XCom's
get/set/clear interface so mapped task instances can push to the
correct entries, and have them pulled correctly by a downstream.

To make the XCom interface easier to use for common scenarios, a
convenience method get_value is added to take a TaskInstanceKey that
automatically performs argument unpacking and call get_one underneath.
This is not done as a get_one overload to simplify the implementation
and typing.

ashb committed:
Update links to new Grid/Graph view instead of relying on redirects (#22167)

Recently we changed the URL strucutre for Graph and Grid (nee Tree)
views, but there were still a few places that used the old names (which
still work thanks to the redirects we have in place.)

This changes the front end to use the new URLs directly.

A few of the places the URL is generated/tweaked via JS, so we needed a
pass a "fake" dag_id -- `$dag_id` to the `url_for()` helper (otherwise
it would blow up) and then replace it directly in the string from JS.

Jorricks committed:
Show DagModel details. (#21868)

* Show DagModel details.

* FIX Invalid extension of list

* FIX Add more items to the excluded list

* FIX Add more items to the excluded list

potiuk committed:
Remove Snowflake limits (#22181)

The limits of snowflake can be removed as the conditions for
all limits are fulfilled now.

This is the last thing to remove before we can enable Python 3.10

dstandish committed:
Use relation in TI join to RTIF (#22159)

Did not include map index; replaced with relation.

potiuk committed:
Avoid trying to kill container when it did not succeed for Docker (#22145)

When container cannot be created in Docker Operator and
exception is raised, the on_kill method attempts to kill the
container but since it is None, it fails with another exception

```
    self.cli.stop(self.container['Id'])
    TypeError: 'NoneType' object is not subscriptable
```

and the original exception is swallowed.

This PR avoids stopping the container if it has not been created.

potiuk committed:
Remove some really old Airflow 1.10 compatibility shims (#22187)

We used some Airflow 1.10 compatibility shims in system
tests and Breeze's entrypoints, in order to be able to run
1.10 still if needed, but the need for that is completely gone
for months now. Time to remove it has long been overdue.

potiuk committed:
Remove extra initialization overhead for mypy/flake (#22183)

The #22127 change introduced a change how execution of docker
commnds was done (due to LD_PRELOAD change) and they started to
use entrypoint_ci, however this caused undesired effect of running
database initialization and printing extra lines which was not needed
and cluttered the output.

This PR introduced SKIP_ENVIRONMENT_INITIALIZATION that (if set to true)
skips the entire initialization of the entrypoint_ci

potiuk committed:
Prepare for Python 3.10 adding (#22075)

We need to merge this change to `main` first before we
enable Python 3.10 in order to make sure our build-image
workflow is ready to build 3.10 images.

potiuk committed:
Remove warnings from kubernetes tests (#22190)

There are warnings generated during K8S tests that there is a
group access to kubeconfig file.

This change makes the kubeconfig only accessible to user.

ephraimbuddy committed:
Bump chart version to 1.6.0-dev (#22192)

Bumping the version to 1.6.0-dev since 1.5.0 is now released

jedcunningham committed:
Move last chart test to `tests` (#22198)

This test was added after the rest of the tests were moved into `tests/charts`.

potiuk committed:
Update instructions of doc updates when removing providers from release (#22199)

When removing providers from release it might be that we remmove
just added provider due to bugs found, in which case more detailewd
instructions on what should be removed from prepared dodcumentation
is needed.

Related to #22197

likecodingloveproblems committed:
Fix typo in mongo.rst (#22211)

anuragbisht12 committed:
Update ``GKEDeleteClusterOperator`, ``GKECreateClusterOperator`` docstrings (#22212)

updated misleading docstrings to use Kubernetes rather than Compute as this takes Kubernetes engine zone as location parameters

wutienyang committed:
fix contributing guide image master -> main (#22216)

Co-authored-by: aiden <wuai@garena.com>

BioGeek committed:
Make code snippet appear again (#22163)

Due to some erroneous extra dots, the reStructuredText got mangled and the code snippet was no longer visible.

stefangsalandy committed:
SFTP docs add example “extras” field using ``key_file``, ``private_key`` and ``host_key`` (#20757)


Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

ulsc committed:
Support Uploading Bigger Files to Google Drive (#22179)

Add `chunk_size` & `resumable` as parameters to `upload_file` method
Change the default `chunk_size` to a clear representation & fix documentation typo

kazanzhy committed:
AWS RDS integration fixes (#22125)

Co-authored-by: Dmytro Kazanzhy <dkazanzhy@demandbase.com>

potiuk committed:
Skip some tests for Databricks from running on Python 3.10 (#22221)

This is a temporary measure until the Databricks SQL library
is released in Python 3.10 - compatible version.

Related to: #22220

ulsc committed:
Change the default `chunk_size` to a clear representation & add documentation (#22222)

alexott committed:
Databricks hook - retry on HTTP Status 429 as well (#21852)

* Databricks hook - retry on HTTP Status 429 as well

this fixes #21559

* Reimplement retries using tenacity

it's now uses exponential backoff by default

vincbeck committed:
Add sample dags and update doc for RedshiftClusterSensor, RedshiftPauseClusterOperator and RedshiftResumeClusterOperator (#22128)

harishkesavarao committed:
Issue 20453 - Fixes the test_http and test_sheets assert calls only (#22104)

megan-parker committed:
Add REST API endpoint for bulk update of DAGs (#19758)

Added endpoint for bulk update of DAGs in the airflow stable API

potiuk committed:
Add the new Airflow Trove Classifier to setup.cfg (#22241)

We have new Trove Classifiers in PyPI for Apache Airflow:

https://github.com/pypa/trove-classifiers/pull/87

This PR adds it for Airflow. The next release of Providers will
add the classifiers for providers.

ephraimbuddy committed:
Use variable for vote end date for helm chart release email (#22239)

The current URL is auto escaped in zsh with backslashes and instead of manually editing this
url, I propose to automate it with environment variables that are not auto escaped by zsh

Also, added a link to the created issue for testing the release in the email body

potiuk committed:
Remove celery from intersphinx mapping temporarily (#22254)

potiuk committed:
Fix assert_queries_count margin to be 0 by default (#22249)

The margin is really only needed for more complex queries. For
most normal queries we should look at the exact value (i.e.
margin should be 0 by default)

zachliu committed:
use different logger to avoid duplicate log entry (#22256)

https://github.com/apache/airflow/pull/22137#issuecomment-1067060405

potiuk committed:
Protect against accidental misuse of XCom.get_value() (#22244)

The XCom.get_value has been added in 2.3.0 and while there are
cases it should be used in the providers when task are mapped,
in order to keep compatibility with earlier versions of Airlfow,
the XCom.get_value() should only be used when ti_key is not None.

We check for the construct used in community providers automatically
and also add a documentation for users who would like to use
dynamic task mapping featuers in their own providers.

Commit:
Add Dataplex operators (#20377)

bbovenzi committed:
Add vertical scrolling to grid view (#22134)

* Add vertical scrolling to grid view

* combine x and y scrolling

* fix zindexes

* comment and linting

* store width as var, move side labels

* lint, remove unused var

ferruzzi committed:
EMR Sample DAG and Docs Update (#22189)

potiuk committed:
Soften the wording in issue templates (#22200)

* Soften the wording in issue templates

Maybe this is my non-native English, but I often feel a bit strange
when reading the issues raised by our users whe they start the
description with:

* "I expected this and that"

For me - this comes from the "What you expected to happen" heading
we have and I have a feeling this might lead to bad wording of the
issues which can lead to unnecessary tensions. "Expectation" is
a bit too strong of a word for the Open-Source project because there
might be many reasons why the "expectation" is not fulfilled and
the user might not be aware, so they might "Expect" something in
a sense of "demanding" it, when there are good reasons things are
how they are.

In most cases people do not read the description or hint we
provided - they stop short by reading the header and base their
answer on the header.

I personally perceive that when you start "I expected this and this",
it feels like "This is obvious it should be like this".

I think changing the heading to "What you think should happen"
might be much better and has much less of the "demand" in it.
Eventually it might lead to people understanding better that what
they raise here is more of a "wish" rather than "expectation" and
that this presents their personal thinking rather than "general
expectation".

Making sure in the heading wording that the raised issue is a bit
"softer" might also set the tone for the follow-up conversation.

* Update .github/ISSUE_TEMPLATE/airflow_bug_report.yml

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

potiuk committed:
Add documentation for Classifier release for March 2022 (#22226)

This is a bulk-release of all providers - it includes
some providers that actually changed, but most of
the providers are just adding the two classifiers:

* Framework :: Apache Airflow
* Framework :: Apache Airflow :: Provider

bbovenzi committed:
Pause autorefresh if scheduler isn't running (#22151)

* pause autorefresh if scheduler isn't running

* fix test

* update jinja if statement

MattTriano committed:
Addressed some issues in the tutorial mentioned in discussion #22233 (#22236)

* Streamlined the tutorial, repaired incorrect logic in the merge, and addressed some concerns in duscussion # 22233

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

mik-laj committed:
Add support for private key in connection for Snowflake (#22266)

potiuk committed:
Fixed issue generation to always include last release (#22265)

dstandish committed:
Enhance `db upgrade` args (#22102)

Make `db upgrade` args more like `db downgrade`.

```
usage: airflow db upgrade [-h] [--from-revision FROM_REVISION] [--from-version FROM_VERSION] [-r REVISION]
                          [-s] [-n VERSION]

Upgrade the schema of the metadata database. To print but not execute commands, use option ``--show-sql-only``. If using options ``--from-revision`` or ``--from-version``, you must also use ``--show-sql-only``, because if actually *running* migrations, we should only migrate from the *current* revision.

optional arguments:
  -h, --help            show this help message and exit
  --from-revision FROM_REVISION
                        (Optional) If generating sql, may supply a *from* revision
  --from-version FROM_VERSION
                        (Optional) If generating sql, may supply a *from* version
  -r REVISION, --revision REVISION
                        (Optional) The airflow revision to upgrade to. Note: must provide either `--revision` or `--version`.
  -s, --show-sql-only   Don't actually run migrations; just print out sql scripts for offline migration. Required if using either `--from-version` or `--from-version`.
  -n VERSION, --version VERSION
                        (Optional) The airflow version to upgrade to. Note: must provide either `--revision` or `--version`.
```

raphaelauv committed:
[FIX] typo doc of gcs operator (#22290)

SamWheating committed:
Set queued_dttm when submitting task to directly to executor (#22259)

steveyz-astro committed:
add a few more fields to the taskinstance finished log message (#22262)

xyu committed:
Filter out default configs when overrides exist. (#21539)

* Filter out default configs when overrides exist.

When sending configs to Airflow workers we materialize a temp config file. In #18772 a feature was added so that `_cmd` generated secrets are not written to the files in some cases instead favoring maintaining the raw `_cmd` settings. Unfortunately during materializing of the configs via `as_dict()` Airflow defaults are generated and materialized as well including defaults for the non `_cmd` versions of some settings. And due to Airflow setting precedence stating bare versions of settings winning over `_cmd` versions it results in `_cmd` settings being discarded:
https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html

This change checks `_cmd`, env, and secrets when materializing configs via `as_dict()` so that if the bare versions of the values is exactly the same as Airflow defaults and we have "hidden" / special versions of these configs that are trying to be set we remove the bare versions so that the correct version can be used.

Fixes: #20092
Related to: #18772 #4050

potiuk committed:
Fail with error when extending image with pip run as root (#22292)

The production docker image installs airflow with --user
flag for `airflow` user and all subsequent image extension
should be done using `airflow` user. It is very easy however,
to run `pip` as root user when you switched temporarily to the
root user for `apt` installation.

This PR makes the accidental `root` user run `pip` fail with
error and redirection to the documentation where it is explained
to the users that they should use `airflow` user for `pip` with
examples.

Fixes: #22250

sunank200 committed:
Add arguments to filter list: start_after_key, from_datetime, to_datetime, object_filter callable (#22231)

Implemented as discussed in [closed PR](https://github.com/apache/airflow/pull/19018).

Add more filter options to list_keys of S3Hook
- `start_after_key`: should return only keys greater than this key
- `from_datetime`: should return only keys with LastModified attr greater than this equal `from_datetime`.
- `to_datetime`: should return only keys with LastModified attr less than this `to_datetime`.
- `object_filter`: Function callable that receives the list of the S3 objects, `from_datetime` and `to_datetime` and returns the List of the matched key.

Add test for the added argument to `list_keys`.

closes: #16627

SamWheating committed:
Masking extras in GET /connections/<connection> endpoint (#22227)

Masking extras in GET /connections/<connection> endpoint

alexott committed:
Operator for updating Databricks Repos (#22278)

* initial version of ReposUpdate operator

Giuzzilla committed:
Added Quick Algorithm Analytics to the list of companies using Apache Airflow (#22279)

dstandish committed:
Enable JSON serialization for connections (#19857)

Previously in general we could only store connections in the Airflow URI format.  With this change we can serialize as JSON.  The Airflow URI format can be very tricky to work with and although we have for some time had a convenience method Connection.get_uri, using JSON is just simpler.

ephraimbuddy committed:
Add test to run offline SQL generation in the CI (#22177)

This ensures our offline migrations command continue to work

talnagar committed:
add an option for run id in the ui trigger screen (#21851)

chenglongyan committed:
Disable default_pool delete on web ui (#21658)

malthe committed:
Add generic connection type (#22310)

See https://github.com/apache/airflow/discussions/20350.

dstandish committed:
Remove RefreshConfiguration workaround for K8s token refreshing (#20759)

A workaround was added (https://github.com/apache/airflow/pull/5731) to handle the refreshing of EKS tokens.  It was necessary because of an upstream bug.  It has since been fixed (https://github.com/kubernetes-client/python-base/commit/70b78cd8488068c014b6d762a0c8d358273865b4) and released in v21.7.0 (https://github.com/kubernetes-client/python/blob/master/CHANGELOG.md#v2170).

pierrejeambrun committed:
Add recipe for BeamRunGoPipelineOperator (#22296)

dstandish committed:
Remove incorrect deprecation warning in secrets backend (#22326)

When the no value is found with `get_conn_value`, the warning was being triggered, even though `get_conn_value` was implemented and just returned no value (cus there wasn't one).

Now we make the logic a little tighter and only raise the dep warning when `get_conn_value` not implemented, which is what we intended to do in the first place.

potiuk committed:
Stronger language about Docker Compose customizability (#22304)

* Stronger language about Docker Compose customizability

Despite our warnings, our users continue treating the Docker
Compose that we exposed as something that should be easy to
extend and customize for their own needs, yet they continue
to struggle with some basic behaviour of containers, Docker Compose
and how they interact. This results in vast space of potential
problems as Docker Compose gives the user a false premise of
something that "just works" where it requires quite a deep
understanding on how it works.

When you get things wrong with Docker Compose, you often end up
with extremely confusing messages, that might suggest that the
problem is with Airflow, but really the problem is with how users
interact with their custom Docker images, registries, pulling,
networking, mounting volumes and plenty other things.

While this is the same with Kubernetes and Helm Chart, Helm Chart makes
it infinitely easier to customize in declarative way (this is what
our values.yaml does) and anything that has not been foreseen by Helm
Chart developers is "hard" by definition.

Docker Compose makes no such distinction. You really can't make Docker
Compose customizable by configuration, and any customization in it
requires modifying the compose file and for people who do not know
what they are doing will eventually lead to errors that they are not
able to diagnose and leads to creation of "Airlfow isssues", where they
should be brought to "Docker Compose" issues.

Example of that is here: https://github.com/apache/airflow/discussions/22301
where there are at least two issues that are not reproducible without
knowing in detail what the user has done, how the image was build
and distributed, and how the docker-compose installation interacted
with them. This leads to a terrible distraction for supporting
users of Airflow as the issues are really Docker Compose issues and
Airflow maintainers should not be involved in solving those.

This PR adds a bit stronger language and statement about the scope
and customizability of the Quick Start Docker Compose of ours. Not
only mentioning "Lack of Production Readiness" but also the
responsibility of the user to understand and diagnose docker compose
errors on their own and setting expectations that issues with Docker
Compose running should be directed elsewhere.

* Update docs/apache-airflow/start/docker.rst

* Update docs/apache-airflow/start/docker.rst

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>

* Update docs/apache-airflow/start/docker.rst

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>

victorphoenix3 committed:
adds ability to pass config params to postgres operator (#21551)

kaxil committed:
Add back celery intersphinx mapping (#22370)

We had disabled this previously in (#22254) but now the website is up on a different domain as listed in https://github.com/celery/celeryproject/issues/51#issuecomment-1072248499

davidhagens committed:
Fix broken links to celery documentation (#22364)

The celery documentation have been moved from https://docs.celeryproject.org/ to https://docs.celeryq.dev/. The old links now refer to a 404 error page, the new links to the actual documentation.

dstandish committed:
Patch sql_alchemy_conn if old postgres scheme used (#22333)

Prior to SqlAlchemy 1.4 the correct scheme for postgres was `postgres+psycopg2` but as of 1.4 it is `postgresql`.  Airflow 2.3 updates SqlAlchemy to 1.4 so unless we patch the config for users (or they update their URIs), upgrading to 2.3 will break.

eladkal committed:
fix deprecation warning in test_default_views.py (#22346)

NomNomGit committed:
Add dataflow_default_options to templated_fields (#22367)

vincbeck committed:
Update sample dag and doc for S3CreateBucketOperator, S3PutBucketTaggingOperator, S3GetBucketTaggingOperator, S3DeleteBucketTaggingOperator, S3DeleteBucketOperator (#22312)

AdamPaslawski committed:
assume file creation time in UTC

AdamPaslawski committed:
merge conflict
